code/datums/components/toggled_clothing.dm

```diff
@@ -0,0 +1,203 @@
+/**
+ * Component which allows clothing to deploy a different kind of clothing onto you.
+ * The simplest example is hooded suits deploying hoods onto your head.
+ */
+/datum/component/toggled_clothing
```

Comment:
This component name is very similar to the existing component  `/datum/component/toggle_icon` (which is its sister component). It might need a rename to better cover the fact that it creates secondary clothing items.